### PR TITLE
[ROOT-10772] Fix build with external libpng

### PR DESCRIPTION
**Pull request by oshadura**
I removed last traces of possible usage of system libpng when build with external libpng. Tested with flags: ` -DCMAKE_PREFIX_PATH=/home/oksana/CERN/libpng/`

Output of making grep of `png` in build directory and ldd output for libAsimage*.so:
https://gist.github.com/oshadura/02284042676515002dbd7383ef5f75c7

### CODE REVIEW

File: cmake/modules/SearchInstalledSoftware.cmake [hunk 1]
```diff
@@ -356,6 +356,9 @@ if(asimage)
   find_Package(PNG)
   if(PNG_FOUND)
     set(ASEXTRA_LIBRARIES ${ASEXTRA_LIBRARIES} ${PNG_LIBRARIES})
+    # Some missing variables needed for external PNG build
+    set(PNG_LIBRARY_RELEASE ${PNG_LIBRARY})
```
Comment (Axel-Naumann) [inline cmake/modules/SearchInstalledSoftware.cmake:360]: Who's using `PNG_LIBRARY_RELEASE`? I see no usage in all of ROOT?

Comment (oshadura) [reply cmake/modules/SearchInstalledSoftware.cmake:360]: It is a bit of "jungles" so when passing non-system png and printing cached CMake variables, I did it to be sure that all that there are no traces of system png library... (if you use system png library, value will be the same).

**Comment by jgrosseo** [comment]
Thanks for the fix!

I imported the patch on top of  v6.20/02, recompiled and to my surprise in my case the libpng still points to the system one:
`libpng15.so.15 => /lib64/libpng15.so.15 (0x00007f99416d0000)`

Log: https://jgrosseo.web.cern.ch/jgrosseo/tmp/log
CMakeCache.txt: https://jgrosseo.web.cern.ch/jgrosseo/tmp/CMakeCache.txt

**Comment by oshadura** [comment]
For Asimage libraries you have now proper support of external PNG library.

@jgrosseo did you check for ldd for `root` executable?

libpng from ldd output of root executable is coming as a dependency from X11 libraries: https://github.com/root-project/root/blob/master/rootx/CMakeLists.txt#L24-L26

We are linking `root` (not root.exe) with next X11 libraries: /usr/lib/libSM.so /usr/lib/libICE.so /usr/lib/libX11.so /usr/lib/libXext.so /usr/lib/libXft.so /usr/lib/libXpm.so
```
 oksana@oksana-dellxps:  ldd /usr/lib/libXft.so 
        linux-vdso.so.1 (0x00007ffecfcf6000)
        libfontconfig.so.1 => /usr/lib/libfontconfig.so.1 (0x00007f9bffd47000)
        libfreetype.so.6 => /usr/lib/libfreetype.so.6 (0x00007f9bffc79000)
        libXrender.so.1 => /usr/lib/libXrender.so.1 (0x00007f9bffc6c000)
        libX11.so.6 => /usr/lib/libX11.so.6 (0x00007f9bffb2c000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f9bff966000)
        libexpat.so.1 => /usr/lib/libexpat.so.1 (0x00007f9bff936000)
        libuuid.so.1 => /usr/lib/libuuid.so.1 (0x00007f9bff92b000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f9bff909000)
        libbz2.so.1.0 => /usr/lib/libbz2.so.1.0 (0x00007f9bff8f6000)
        libpng16.so.16 => /usr/lib/libpng16.so.16 (0x00007f9bff8be000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f9bff8a4000)
        libharfbuzz.so.0 => /usr/lib/libharfbuzz.so.0 (0x00007f9bff7a0000)
        libxcb.so.1 => /usr/lib/libxcb.so.1 (0x00007f9bff774000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f9bff76f000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007f9bffdf1000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f9bff629000)
        libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0x00007f9bff501000)
        libgraphite2.so.3 => /usr/lib/libgraphite2.so.3 (0x00007f9bff4dc000)
        libXau.so.6 => /usr/lib/libXau.so.6 (0x00007f9bff4d7000)
        libXdmcp.so.6 => /usr/lib/libXdmcp.so.6 (0x00007f9bff4cd000)
        libpcre.so.1 => /usr/lib/libpcre.so.1 (0x00007f9bff45b000)

``` 
Here I am not sure how I can help you, only solution for you is to disable x11.

**Comment by jgrosseo** [comment]
Hi,
Thanks - this clarifies. Indeed, I was checking root, for root.exe, it is not there. 

For libASImage, I get actually both:
`libpng16.so.16 => /tmp/o2/sw/slc7_x86-64/libpng/v1.6.34-58/lib/libpng16.so.16 (0x00007f1351718000)
libpng15.so.15 => /lib64/libpng15.so.15 (0x00007f134d24e000)`

Is this expected?

**Comment by oshadura** [comment]
Can you try to make "clean build" with patch on the top? I see you still have mixed env variables.
My output from "clean build"

```   
        linux-vdso.so.1 (0x00007ffd0a4fe000)
        libgif.so.7 => /usr/lib/libgif.so.7 (0x00007f1ce73db000)
        libtiff.so.5 => /usr/lib/libtiff.so.5 (0x00007f1ce734c000)
        libpng16.so.16 => /home/oksana/CERN/libpng/lib/libpng16.so.16 (0x00007f1ce7305000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f1ce72eb000)
        libjpeg.so.8 => /usr/lib/libjpeg.so.8 (0x00007f1ce7254000)
        libfreetype.so.6 => /usr/lib/libfreetype.so.6 (0x00007f1ce7186000)
        libSM.so.6 => /usr/lib/libSM.so.6 (0x00007f1ce717a000)
        libICE.so.6 => /usr/lib/libICE.so.6 (0x00007f1ce715d000)
        libX11.so.6 => /usr/lib/libX11.so.6 (0x00007f1ce701d000)
        libXext.so.6 => /usr/lib/libXext.so.6 (0x00007f1ce7008000)
        libGraf.so => /home/oksana/CERN_sources/root/builds/lib/libGraf.so (0x00007f1ce6e94000)
        libHist.so => /home/oksana/CERN_sources/root/builds/lib/libHist.so (0x00007f1ce6a7f000)
        libMatrix.so => /home/oksana/CERN_sources/root/builds/lib/libMatrix.so (0x00007f1ce68ff000)
        libMathCore.so => /home/oksana/CERN_sources/root/builds/lib/libMathCore.so (0x00007f1ce66e6000)
        libImt.so => /home/oksana/CERN_sources/root/builds/lib/libImt.so (0x00007f1ce66d4000)
        libRIO.so => /home/oksana/CERN_sources/root/builds/lib/libRIO.so (0x00007f1ce62fd000)
        libThread.so => /home/oksana/CERN_sources/root/builds/lib/libThread.so (0x00007f1ce62a7000)
        libCore.so => /home/oksana/CERN_sources/root/builds/lib/libCore.so (0x00007f1ce5e2b000)
        libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f1ce5e07000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f1ce5c1e000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f1ce5ad8000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f1ce5abe000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f1ce58f8000)
        libzstd.so.1 => /usr/lib/libzstd.so.1 (0x00007f1ce584e000)
        liblzma.so.5 => /usr/lib/liblzma.so.5 (0x00007f1ce5824000)
        libbz2.so.1.0 => /usr/lib/libbz2.so.1.0 (0x00007f1ce5811000)
        libharfbuzz.so.0 => /usr/lib/libharfbuzz.so.0 (0x00007f1ce570d000)
        libuuid.so.1 => /usr/lib/libuuid.so.1 (0x00007f1ce5704000)
        libxcb.so.1 => /usr/lib/libxcb.so.1 (0x00007f1ce56da000)
        libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f1ce56d5000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007f1ce74f5000)
        libtbb.so.2 => /usr/lib/libtbb.so.2 (0x00007f1ce568d000)
        libpcre.so.1 => /usr/lib/libpcre.so.1 (0x00007f1ce561b000)
        libxxhash.so.0 => /usr/lib/libxxhash.so.0 (0x00007f1ce5611000)
        liblz4.so.1 => /usr/lib/liblz4.so.1 (0x00007f1ce55ef000)
        libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0x00007f1ce54c5000)
        libgraphite2.so.3 => /usr/lib/libgraphite2.so.3 (0x00007f1ce54a0000)
        libXau.so.6 => /usr/lib/libXau.so.6 (0x00007f1ce549b000)
        libXdmcp.so.6 => /usr/lib/libXdmcp.so.6 (0x00007f1ce5493000)
        librt.so.1 => /usr/lib/librt.so.1 (0x00007f1ce5488000)

```

**Comment by jgrosseo** [comment]
I had thought I made a clean build, but maybe I made a mistake. Unfortunately, I lost the docker in which I had started from a clean CC7 image. 
@ktf will import your patch in our build system to make a full test.

**Pull request merged**